modules/UpgradeWizard/uw_utils.php

```diff
@@ -2248,42 +2256,54 @@ function unlinkUploadFiles()
 //	}
 }
 
+/**
+ * Recursively deletes a directory tree.
+ *
+ * @param string $folder         The directory path.
+ * @param bool   $keepRootFolder Whether to keep the top-level folder.
+ *
+ * @return bool TRUE on success, otherwise FALSE.
+ */
+function deleteTree($folder, $keepRootFolder = false) {
+	// Handle bad arguments.
+	if (empty($folder) || !file_exists($folder)) {
+		return true; // No such file/folder exists.
+	} elseif (is_file($folder) || is_link($folder)) {
+		return @unlink($folder); // Delete file/link.
+	}
+
+	// Delete all children.
+	$files = new \RecursiveIteratorIterator(
+		new \RecursiveDirectoryIterator($folder, \RecursiveDirectoryIterator::SKIP_DOTS),
+		\RecursiveIteratorIterator::CHILD_FIRST
+	);
+
+	foreach ($files as $fileinfo) {
+		$action = ($fileinfo->isDir() ? 'rmdir' : 'unlink');
+		if (!@$action($fileinfo->getRealPath())) {
+			return false; // Abort due to the failure.
+		}
+	}
+
+	// Delete the root folder itself?
+	return (!$keepRootFolder ? @rmdir($folder) : true);
+}
+
 /**
  * deletes files created by unzipping a package
  */
 function unlinkUWTempFiles()
 {
-    global $sugar_config;
     global $path;
+    
+    list(/* ignore first element */, $tempDir) = getUWDirs();
+    deleteTree($tempDir, true);
 
-    logThis('at unlinkUWTempFiles()');
-    $tempDir='';
-    list($upgDir, $tempDir) = getUWDirs();
-
-    if (file_exists($tempDir) && is_dir($tempDir)) {
-        $files = findAllFiles($tempDir, array(), false);
-        rsort($files);
-        foreach ($files as $file) {
-            if (!is_dir($file)) {
-                //logThis('unlinking ['.$file.']', $path);
-                @unlink($file);
-            }
-        }
-        // now do dirs
-        $files = findAllFiles($tempDir, array(), true);
-        foreach ($files as $dir) {
-            if (is_dir($dir)) {
-                //logThis('removing dir ['.$dir.']', $path);
-                @rmdir($dir);
-            }
-        }
-        $cacheFile = sugar_cached("modules/UpgradeWizard/_persistence.php");
-        if (is_file($cacheFile)) {
-            logThis("Unlinking Upgrade cache file: '_persistence.php'", $path);
-            @unlink($cacheFile);
-        }
+    $cacheFile = sugar_cached("modules/UpgradeWizard/_persistence.php");
```

Comment:
Single quotes where possible:
```suggestion
    $cacheFile = sugar_cached('modules/UpgradeWizard/_persistence.php');
```